src/poly1305.rs

```diff
@@ -16,31 +16,86 @@
 // TODO: enforce maximum input length.
 
 use {c, chacha, constant_time, error};
-
-pub struct SigningContext {
-    state: State,
-}
+use polyfill::slice::u32_from_le_u8;
+use std::cmp::min;
+use std::mem::align_of;
+use std::os::raw::c_void;
```

Comment:
1. Please use `core` instead of `std` whenever possible.
2. `use core;` and then refer to these qualified as `core::cmp::min()`, etc.